src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -49,9 +49,28 @@ private static RuntimeType GetRuntimeTypeFromHandleSlow(IntPtr handle)
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern RuntimeType? GetRuntimeTypeFromHandleIfExists(IntPtr handle);
 
-        private static RuntimeType GetRuntimeTypeFromHandle(IntPtr handle)
+        // implementation of CORINFO_HELP_GETSYNCFROMCLASSHANDLE, CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE, CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPEHANDLE
+        internal static unsafe RuntimeType GetRuntimeTypeFromHandle(IntPtr handle)
```

Comment:
This is 1.5x microbenchmark perf regression: https://github.com/EgorBot/runtime-utils/issues/225#issuecomment-2574011409 . Could you please take a look at the disassembly to see whether we can do something about it?

Reply:
Yep. I'm looking at it.